frontend/src/features/projects/common/ProjectRouter.tsx

```diff
@@ -32,33 +34,26 @@ const ProjectRouter = ({ location }: { match: Match; location: Location }) => {
   const query = location?.search ?? {};
   const formikRef = useRef<FormikValues>();
   const projectNumber = queryString.parse(query).projectNumber;
+  const { project } = useProject();
   const dispatch = useDispatch();
   const getProjectRequest = useSelector<RootState, IGenericNetworkAction>(
-    state => (state.network as any)[ProjectActions.GET_PROJECT] as any,
+    state => (state.network as any)[ProjectActions.GET_PROJECT_WORKFLOW] as any,
   );
-  const project: any = useSelector<RootState, IProjectWrapper>(state => state.project).project;
   useEffect(() => {
     if (projectNumber !== null && projectNumber !== undefined) {
       dispatch(clearProject());
-      dispatch(fetchProject(projectNumber as string));
+      (dispatch(fetchProject(projectNumber as string)) as any).then((project: IProject) => {
+        dispatch(fetchProjectWorkflow(project?.workflowCode));
+      });
     }
   }, [dispatch, projectNumber]);
 
-  const workflowStatuses = useSelector<RootState, IStatus[]>(state => state.projectWorkflow as any);
-  const workflowStatusesLength = workflowStatuses?.length;
-  const workflowCode = workflowStatuses?.length > 0 ? workflowStatuses[0].workflowCode : undefined;
-  useEffect(() => {
-    if (workflowStatusesLength === 0 || workflowCode !== project?.workflowCode) {
-      dispatch(fetchProjectWorkflow(project?.workflowCode));
-    }
-  }, [dispatch, project, workflowCode, workflowStatusesLength]);
-
   if (projectNumber !== null && projectNumber !== undefined && getProjectRequest?.error) {
     throw Error(`Unable to load project number ${projectNumber}`);
   }
   return (
     <>
-      {getProjectRequest?.isFetching !== true ? (
+      {getProjectRequest?.isFetching === false && projectNumber === project.projectNumber ? (
```

Comment:
also ensure that the project that we have stored in redux matches the projectnumber in the path.